src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionExpirationQueueConsumerTests.kt

```diff
@@ -3258,4 +3282,333 @@ class TransactionExpirationQueueConsumerTests {
+            .toMono()
+        }
+      given(
+          deadLetterTracedQueueAsyncClient.sendAndTraceDeadLetterQueueEvent(
+            capture(binaryDataCaptor), any()))
+        .willReturn(mono {})
+
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.right(
+            QueueEvent(
+              events.filterIsInstance<TransactionExpiredEvent>().first(), MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsRefundedEventStoreRepository, times(2)).save(any())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundRequestedEvent>()
+          .count())
+      assertEquals(
+        1,
+        transactionRefundEventStoreCaptor.allValues
+          .filterIsInstance<TransactionRefundedEvent>()
+          .count())
+      verify(deadLetterTracedQueueAsyncClient, times(0))
+        .sendAndTraceDeadLetterQueueEvent(any(), any())
+    }
+
+    @Test
+    fun `messageReceiver should not perform REFUND when NPG has not authorized the transaction`() {
+      val events =
+        listOf(
+          transactionActivateEvent(
+            NpgTransactionGatewayActivationData(
+              UUID.randomUUID().toString(), UUID.randomUUID().toString())),
+          transactionAuthorizationRequestedEvent(
+            TransactionAuthorizationRequestData.PaymentGateway.NPG,
+            NpgTransactionGatewayAuthorizationRequestedData(
+              URI.create("http://localhost"),
+              "MC",
+              UUID.randomUUID().toString(),
+              UUID.randomUUID().toString())))
+      setupTransactionStorageMock(events)
+      given(checkpointer.success()).willReturn(Mono.empty())
+      given { authorizationStateRetrieverService.getStateNpg(any(), any(), any(), any(), any()) }
+        .willAnswer {
+          StateResponseDto()
+            .state(WorkflowStateDto.PAYMENT_COMPLETE)
+            .operation(
+              OperationDto()
+                .orderId(UUID.randomUUID().toString())
+                .operationResult(OperationResultDto.CANCELED)
+                .paymentEndToEndId(UUID.randomUUID().toString())
+                .operationTime(ZonedDateTime.now().toString()))
+            .toMono()
+        }
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.right(
+            QueueEvent(
+              transactionExpiredEvent(TransactionStatusDto.AUTHORIZATION_REQUESTED),
+              MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsExpiredEventStoreRepository, times(1)).save(any())
+      verify(transactionsRefundedEventStoreRepository, times(0)).save(any())
+      verify(refundService, times(0)).requestNpgRefund(any(), any(), any(), any(), any(), any())
```

Comment:
add verification against dead lettere queue not being written

Reply:
Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/255/commits/a03dde092c41a59ff7dcefdfb6f0a5b0a4585274